### PR TITLE
POSIX shell compliance

### PR DESCRIPTION
**Pull request by r2evans**
Remove bashisms so it can be run in a posix-shell environment. The impetus for this is running `pushbullet-bash` on a [DD-WRT](http://www.dd-wrt.com/site/index) (customizable wireless router) instance with `/bin/sh` only, `bash` is often not installed.

### CODE REVIEW

**Comment by fbartels** [comment]
Hi @r2evans, 

I've adopted your changes a bit and pushed the current code to https://github.com/Red5d/pushbullet-bash/tree/r2evans-posixShell. So far the changes look good it would be ok to merge from my side.

@Red5d since this diverts a bit from "This is intended to be a pure Bash utility with no non-Bash dependencies." to posix compatibility I did not want to merge it right away. Are you ok with the merge?

**Comment by r2evans** [comment]
@fbartels @Red5d 

I updated the branch to reflect the merge of errorCodes, and this includes a change to README.

File: pushbullet [hunk 1]
```diff
@@ -106,28 +119,57 @@ function getactivepushes () {
...
-		# suppress error when calling break outside of loop
-		break 2&> /dev/null
-	elif [[ "$res" != "created" ]] && [[ ! "$1" == "{}" ]]; then
-		err "Error submitting the request. The error message was:" "$1"
-		exit 1
-	fi
+	curlretcode=${2:-$ERR_UNK}
+	checkCurlReturnCode "$curlretcode"
+	case "$1" in
+		*"The param 'channel_tag' has an invalid value."*)
```
Comment (fbartels) [inline pushbullet]: while this is a correct translation of the code, I think this is not that the original intention was. I think instead this should be
```
case ...
*"The param 'channel_tag' has an invalid value."* | *"The param 'device_iden' has an invalid value."*)
...

**Comment by Red5d** [comment]
Merged. While it is a slight departure from being "pure Bash", it fits the intent of not requiring additional dependencies and can be run "as is".

**Pull request merged**